src/main/java/space/npstr/magma/processing/PacketProvider.java

```diff
@@ -0,0 +1,133 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package space.npstr.magma.processing;
+
+import com.sun.jna.ptr.PointerByReference;
+import net.dv8tion.jda.core.audio.AudioPacket;
+import net.dv8tion.jda.core.audio.AudioSendHandler;
+import net.dv8tion.jda.core.audio.factory.IPacketProvider;
+import net.dv8tion.jda.core.audio.hooks.ConnectionStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import space.npstr.magma.EncryptionMode;
+import space.npstr.magma.connections.AudioConnection;
+
+import javax.annotation.Nullable;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetSocketAddress;
+import java.util.function.Supplier;
+
+/**
+ * Created by napster on 23.06.18.
+ */
+public class PacketProvider implements IPacketProvider {
+
+    private static final Logger log = LoggerFactory.getLogger(PacketProvider.class);
+    private static final String INFORMATION_NOT_AVAILABLE = "This information is not available";
+
+    private final AudioConnection audioConnection;
+    private final Supplier<Long> nonceSupplier;
+
+    private char seq = 0;           //Sequence of audio packets. Used to determine the order of the packets.
+    private int timestamp = 0;      //Used to sync up our packets within the same timeframe of other people talking.
+
+    public PacketProvider(final AudioConnection audioConnection, final Supplier<Long> nonceSupplier) {
+        this.audioConnection = audioConnection;
+        this.nonceSupplier = nonceSupplier;
+    }
+
+    @Override
+    public String getIdentifier() {
+        throw new UnsupportedOperationException(INFORMATION_NOT_AVAILABLE);
+    }
+
+    @Override
+    public String getConnectedChannel() {
+        throw new UnsupportedOperationException(INFORMATION_NOT_AVAILABLE);
+    }
+
+    @Override
+    public DatagramSocket getUdpSocket() {
+        throw new UnsupportedOperationException(INFORMATION_NOT_AVAILABLE);
+    }
+
+    @Override
+    public void onConnectionError(final ConnectionStatus status) {
+        throw new UnsupportedOperationException("Connection error, on a udp connection...that's not a real thing.");
+    }
+
+    @Override
+    public void onConnectionLost() {
+        throw new UnsupportedOperationException("Connection lost, on a udp connection...that's not a real thing.");
+    }
+
+    //realistically, this is the only thing that is ever called by the NativeAudioSystem
+    @Nullable
+    @Override
+    public DatagramPacket getNextPacket(final boolean changeTalking) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 24 to the 15 allowed. ([squid:S3776](https://sonarcloud.io/organizations/napstr-github/rules#rule_key=squid%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=space.npstr.magma%3Amagma&pullRequest=6&issues=AWQs3zm6bXXTsx7OQREk&open=AWQs3zm6bXXTsx7OQREk)

[ISSUE_KEY]: AWQs3zm6bXXTsx7OQREk